### PR TITLE
Be more lenient with errors in heartbeats

### PR DESCRIPTION
**Pull request by stevekuznetsov**
When the boskosctl command fails to send one heartbeat, it is not a
catastrophic error as we send heartbeats much more frequently than
strictly necessary to retain our lease. This allows us to keep an error
budget and exit only when exceeding it, which is valuable as transitive
network issues should not crash the heartbeat command.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/assign @krzyzacy @sebastienvas

### CODE REVIEW

**Comment by wking** [comment]
This isn't really specific to heartbeats is it?  Should we build a retry handler into `k8s.io/test-infra/boskos/client`

**Comment by stevekuznetsov** [comment]
We could add retries there, too, I guess

File: boskos/cmd/cli/cli.go [hunk 1]
```diff
@@ -284,15 +285,21 @@ Examples:
...
 						fmt.Fprintf(cmd.ErrOrStderr(), "failed to send heartbeat for resource %q: %v\n", resource.Name, err)
-						exit(1)
-						return true
+						failures++
+						if failures > options.heartbeat.retries {
+							exit(1)
+							return true
+						}
+					} else {
+						fmt.Fprintf(cmd.OutOrStdout(), "heartbeat sent for resource %q\n", resource.Name)
```
Comment (alvaroaleman) [inline boskos/cmd/cli/cli.go]: Would it make sense to set `failures` back to 0 on success?

Comment (stevekuznetsov) [reply boskos/cmd/cli/cli.go]: Could do that as well, too. I was just looking for a simple approach :)

**Comment by stevekuznetsov** [comment]
@krzyzacy @sebastienvas so forgetting other, more complex ideas ... how do we feel about this PR?

File: boskos/cmd/cli/cli.go (hunk 1 above)
Comment (sebastienvas) [reply boskos/cmd/cli/cli.go]: +1

**Comment by stevekuznetsov** [comment]
Updated to reset on success @alvaroaleman @sebastienvas

File: go.mod [hunk 2]
```diff
@@ -123,3 +123,5 @@ require (
 	sigs.k8s.io/yaml v1.1.0
 	vbom.ml/util v0.0.0-20170409195630-256737ac55c4
 )
+
+go 1.13
```
Comment (alvaroaleman) [inline go.mod]: Do we want to check this in?

Comment (stevekuznetsov) [reply go.mod]: ah yikes, good catch

**Comment by stevekuznetsov** [comment]
@alvaroaleman @sebastienvas @krzyzacy any final thoughts?

File: boskos/cmd/cli/cli.go [hunk 3]
```diff
@@ -284,15 +285,22 @@ Examples:
...
 				case <-tick:
 					if err := options.c.Update(resource.Name, resource.State, resource.UserData); err != nil {
 						fmt.Fprintf(cmd.ErrOrStderr(), "failed to send heartbeat for resource %q: %v\n", resource.Name, err)
-						exit(1)
-						return true
+						failures++
+						if failures > options.heartbeat.retries {
+							exit(1)
+							return true
+						}
```
Comment (wking) [inline boskos/cmd/cli/cli.go]: Did we want a back off sleep here?

**Pull request merged**